app/models/fe/page.rb

```diff
@@ -53,7 +53,11 @@ class Page < ActiveRecord::Base
     # end
 
     def has_questions?
-      questions.present? || question_grids.present? || question_grid_with_totals.present?
+      all_questions.present?
+    end
+
+    def all_questions
+      all_elements.where("kind NOT IN('Fe::Paragraph', 'Fe::Section', 'Fe::QuestionGrid', 'Fe::QuestionGridWithTotal')")
```

Comment:
This list of element types shows up a few different places in the code. Let's extract it out into a constant.